server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -895,6 +897,31 @@ ClusterState improveConfiguration(ClusterState clusterState) {
         return clusterState;
     }
 
+    /*
+    * Valid Voting Configuration Exclusion state criteria:
+    * 1. Every voting config exclusion with an ID of _absent_ should not match any nodes currently in the cluster by name
+    * 2. Every voting config exclusion with a name of _absent_ should not match any nodes currently in the cluster by ID
+     */
+    private boolean validVotingConfigExclusionState(ClusterState clusterState) {
```

Comment:
Suggest making this non-private and static so you can test it directly, rather than indirectly by catching assertions in tests.